ci/desktop.yml

```diff
@@ -0,0 +1,108 @@
+# To learn how to migrate your existing WPF application to .NET Core,
+# refer to https://docs.microsoft.com/en-us/dotnet/desktop-wpf/migration/convert-project-from-net-framework
+#
+# To configure this workflow:
+#
+# 1. Configure environment variables
+# GitHub sets default environment variables for every workflow run.  Replace the variables relative to your
+# project in the "env" section below.
+# 
+# 2. Signing
+# Generate a signing certificate in the Windows Application Packaging Project or add an existing signing certificate
+# to the project. Next, use PowerShell to encode the .pfx file using Base64 encoding by running the following Powershell
+# script to generate the output file.
+# 
+# $pfx_cert = Get-Content '.\SigningCertificate.pfx' -Encoding Byte
+# [System.Convert]::ToBase64String($pfx_cert) | Out-File 'SigningCertificate_Encoded.txt'
+#
+# Open the output file, SigningCertificate_Encoded.txt, and copy the string inside. Then, add the string to the repo
+# as a GitHub secret and name it "Base64_Encoded_Pfx."
+# For more information on how to configure your signing certificate for this workflow, refer
+# to https://github.com/microsoft/github-actions-for-desktop-apps#signing
+#
+# Finally, add the signing certificate password to the repo as a secret and name it "Pfx_Key."
+# See "Build the Windows Application Packaging project" below to see how the secret is used.
+#
+# For more information on GitHub Actions, refer to https://github.com/features/actions
+# For a complete CI/CD sample to get started with GitHub Action workflows for Desktop Applications,
+# refer to https://github.com/microsoft/github-actions-for-desktop-apps
+
+name:  Wpf .NET Core
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+
+    runs-on: windows-latest
+
+    env:
+      Appx_Bundle: Always                                       # Creates an .msixbundle/.appxbundle with the .msix/.appx files for the platform specified.
+      App_Packages_Directory: AppPackages                       # The name of the directory that will contain the application packages. The default is "AppPackages."
+      Appx_Bundle_Platforms: x86|x64                            # Include x86 and x64 platforms in the bundle. Possible values: x86|x64|ARM
+      Appx_Package_Build_Mode: StoreUpload                      # Generates the .msixupload/.appxupload file and the _Test folder for sideloading.
+      Signing_Certificate: GitHubActionsDemo.pfx                # The name of the .pfx that will be created by decoding the Base 64 secret and consumed during the packaging step.
+      Solution_Name: your-solution-name                         # Replace with your solution name, i.e. MyWpfApp.sln.
+      Test_Project_Path: your-test-project-path                 # Replace with the path to your test project, i.e. MyWpfApp.Tests\MyWpfApp.Tests.csproj.
```

Comment:
Wow, this is a _lot_ of settings.  Is all of this necessary?  For example: why would you want to set up `Test_Project_Path` here?  What if you had two test projects not one?  Wouldn't just running `dotnet test` support multiple test projects _and_ reduce the amount of configuration?

These workflows aren't necessarily meant to solve _every problem_, they're to help people get started and understand how GitHub Actions works.  The more that we can automate, the less that we require people to change, and the fewer options that we have, the better.  People can grow-up into their own scenarios once we've given them the basics.

Reply:
Edward, thanks for your comment!  

I can reduce the number of environment variables considerably by hardcoding a few of the settings.  That should makes things less complicated.

When it comes to the test project, unfortunately, when you run "dotnet test" at the solution level, Nuget Restore tries to restore all the projects in the solution, including the Windows Application Project.  The result is a warning and and error, right off the bat.   It doesn't affect testing.  But it doesn't look nice either and may confound our users.

![dotnetTest](https://user-images.githubusercontent.com/5585130/78688100-e473bc00-78c2-11ea-949a-8e1549ac97f3.png)


Notice when we specify the test project, that doesn't happen.
![dotNetTestProject](https://user-images.githubusercontent.com/5585130/78688107-e6d61600-78c2-11ea-9fc4-363d108dfe84.png)

What do you think?  Is that too confusing for users?

I'll work on removing some of the environment variables and address Srdjan's feedback.

Thanks again!

Reply:
Aha!  I see!  Thanks for the additional information.  Is there an example project that I can use to understand this better?  I haven't written any WPF, my .NET UI experience is limited to WinForms.

Reply:
Take a look here:    https://github.com/microsoft/github-actions-for-desktop-apps

I based this workflow on the CI pipeline I developed for this sample app.  I removed a few unnecessary things, such as versioning and simplified how the project is built and what options are available to customers out of the box.

Please let me know if you have questions.

Reply:
@ethomson Hi!  I'm just checking up on this.  Is there anything else you need?